src/scheduler/fifo.c

```diff
@@ -1242,17 +1242,12 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int throughput, schd_e
 					snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
 					schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
 				}
-				if (throughput)
-					rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-				else
-					rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 			}
-		} else {
-			if (throughput)
-				rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-			else
-				rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 		}
+		if (rc > 0 && throughput)
+			rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
+		else
```

Comment:
Well we do bail out and not run the job if there is an error while setting walltime of a STF job. the block checking for return code its the error message from the socket descriptor and puts it in the error structure before returning.
Is this what you were asking?